### DISCUSSION TITLE
How to add new Protocol for "on behalf of a user" authentication method

### DISCUSSION THREAD

**Discussion by nkgiovannivl** [O365/python-o365#595] [Q&A]
Hi All,

How to add new Protocol for "on behalf of a user" authentication method and scopes?

Looking inside the guide, the https://github.com/O365/python-o365/blob/master/O365/connection.py#L34 most be modified or there is a way to change the configuration?

PowerBi protocol_url: https://analysis.windows.net/powerbi/api/

Some scope examples for creation on "scope helpers":
"https://analysis.windows.net/powerbi/api/Dataset.Read.All",
"https://analysis.windows.net/powerbi/api/Dataset.ReadWrite.All",
"https://analysis.windows.net/powerbi/api/Group.Read",
"https://analysis.windows.net/powerbi/api/Group.Read.All",        
"https://analysis.windows.net/powerbi/api/Workspace.Read.All",
"https://analysis.windows.net/powerbi/api/Workspace.ReadWrite.All"

Thanks!

**Comment by alejcas** [comment]
You should first implement a protocol and then add a new class object to work with the api.
